### ISSUE TITLE
Exceptions while downloading proving key on Mac OS

### ISSUE THREAD

**Issue by bergr01** [BTCPrivate/bitcoin-private-full-node-wallet#34]
<img width="981" alt="screen shot 2018-03-05 at 11 13 36 pm" src="https://user-images.githubusercontent.com/4503791/36975304-cea38cce-20cd-11e8-8fd5-333b304ee8f8.png">
<img width="996" alt="screen shot 2018-03-05 at 11 25 13 pm" src="https://user-images.githubusercontent.com/4503791/36975340-e379c9a6-20cd-11e8-9262-afa4951c4eb0.png">

see exception stack traces attached.

**Comment by fokep** [comment]
Could you attach the debug log? Should be here:

`~/Library/Application Support/BTCPrivateDesktopWallet/BitcoinPrivateGUIWallet_2018_5_debug.log`

May be a duplicate of #9 / #22

**Comment by bergr01** [comment]
Here is the entire content of the log file:

[main] [Mon Mar 05 23:09:53 AEDT 2018] [INFO] btcprivate.conf (/Users/myusername/Library/Application Support/BTCPrivate/btcprivate.conf) does not exist. It will be created with default settings. 
[main] [Mon Mar 05 23:09:53 AEDT 2018] [INFO] Bitcoin Private Full-Node Desktop Wallet (GUI, made in Java & Swing) 
[main] [Mon Mar 05 23:09:53 AEDT 2018] [INFO] OS: Mac OS X = MAC_OS 
[main] [Mon Mar 05 23:09:53 AEDT 2018] [INFO] Current directory: /Users/gregberesnev/Downloads/BitcoinPrivateDesktopWallet.app/Contents 
[main] [Mon Mar 05 23:09:53 AEDT 2018] [INFO] Class path: /Users/myusername/Downloads/BitcoinPrivateDesktopWallet.app/Contents/Java/Classes:/Users/myusername/Downloads/BitcoinPrivateDesktopWallet.app/Contents/Java/BitcoinPrivateDesktopWallet.jar 
[main] [Mon Mar 05 23:09:53 AEDT 2018] [INFO] Environment PATH: /usr/bin:/bin:/usr/sbin:/sbin 
[main] [Mon Mar 05 23:09:53 AEDT 2018] [INFO] Using Bitcoin Private daemon and rpc tools: btcpd: /Users/myusername/Downloads/BitcoinPrivateDesktopWallet.app/Contents/btcpd, btcp-cli: /Users/myusername/Downloads/BitcoinPrivateDesktopWallet.app/Contents/btcp-cli 
[main] [Mon Mar 05 23:09:53 AEDT 2018] [INFO] btcpd is not running at the moment or has not started/synchronized 100% - showing splash... 
[main] [Mon Mar 05 23:16:14 AEDT 2018] [INFO] Bitcoin Private Full-Node Desktop Wallet (GUI, made in Java & Swing) 
[main] [Mon Mar 05 23:16:14 AEDT 2018] [INFO] OS: Mac OS X = MAC_OS 
[main] [Mon Mar 05 23:16:14 AEDT 2018] [INFO] Current directory: /Users/myusername/Downloads/BitcoinPrivateDesktopWallet.app/Contents 
[main] [Mon Mar 05 23:16:14 AEDT 2018] [INFO] Class path: /Users/myusername/Downloads/BitcoinPrivateDesktopWallet.app/Contents/Java/Classes:/Users/myusername/Downloads/BitcoinPrivateDesktopWallet.app/Contents/Java/BitcoinPrivateDesktopWallet.jar 
[main] [Mon Mar 05 23:16:14 AEDT 2018] [INFO] Environment PATH: /usr/bin:/bin:/usr/sbin:/sbin 
[main] [Mon Mar 05 23:16:14 AEDT 2018] [INFO] Using Bitcoin Private daemon and rpc tools: btcpd: /Users/myusername/Downloads/BitcoinPrivateDesktopWallet.app/Contents/btcpd, btcp-cli: /Users/myusername/Downloads/BitcoinPrivateDesktopWallet.app/Contents/btcp-cli 
[main] [Mon Mar 05 23:16:14 AEDT 2018] [INFO] btcpd is not running at the moment or has not started/synchronized 100% - showing splash...

What was the resolution for #9 / #22 exactly, it wasn't clear to me from opening those tickets?

**Comment by fokep** [comment]
Try to run `btcpd -noindex` before starting the wallet

**Comment by bergr01** [comment]
If by running btcpd you mean stepping into '/Users/myusername/Downloads/BitcoinPrivateDesktopWallet.app/Contents' and running ./btcpd 
then that's what I got:

./btcpd -noindex

dyld: Symbol not found: _clock_gettime
  Referenced from: /Users/myusername/Downloads/BitcoinPrivateDesktopWallet.app/Contents/./btcpd
  Expected in: /usr/lib/libSystem.B.dylib
 in /Users/myusername/Downloads/BitcoinPrivateDesktopWallet.app/Contents/./btcpd
Trace/BPT trap: 5

**Comment by bergr01** [comment]
Googling for that error message yields results such as https://github.com/pocoproject/poco/issues/1453

**Comment by fokep** [comment]
Ah! Updating to Mac OS > 10.12 should fix it then?On Mar 6, 2018 01:50, bergr01 <notifications@github.com> wrote:Googling for that error message yields results such as pocoproject/poco#1453

**Comment by bergr01** [comment]
You want me to update to a newer version of my OS to make your wallet work? Is there any alternative, like you making wallet backward compatible with El Capitan or some other wallet I can use for the same purpose?

**Comment by bergr01** [comment]
Ended up installing the Windows version of the https://github.com/BTCPrivate/electrum-btcp/releases wallet and things worked okay there.

**Comment by fokep** [comment]
I don't understand why you wouldn't upgrade independent of that but it is of course your choice to run whatever you wish. A lot of people are still hanging on to Windows 7 as well.

Making the full node wallet backwards compatible with older OS's will not be a priority but it may very well be possible that it can be done sometime in the future. It's tricky and it requires a lot of work to make software that is compatible on all platforms including old/obsolete/depreciated OS versions. Right now the primary focus is to get everything up and running for people to claim their BTCP.
